tests/test_request_params.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
```

Comment:
we already have a file for these test, called tests/test_query_params.py. could you move the tests there?

also you can probably check there how those methods are tested